### ISSUE TITLE
"INVALID GPU OR INVALID FILES IN FOLDER - FAILED"

### ISSUE THREAD

**Issue by kriebe** [upscayl/upscayl#669]
### Checklist

- [X] I have checked that this issue isn't addressed in [the FAQ](https://github.com/upscayl/upscayl#-faq).
- [X] I have checked that this issue isn't addressed in any other issue or pull request.
- [ ] I have tried [setting my GPU ID](https://github.com/upscayl/upscayl/wiki/Guide#gpu-id).
- [ ] I am using Windows and I've tried [common troubleshooting steps](https://github.com/upscayl/upscayl/wiki/Troubleshooting).

### Describe the Bug

I started with a much larger folder of images to upscale, but was able to reproduce with just 4 images in a folder.  With these 4 images, when I attempted to do a batch upscale, as soon as I hit the Upscayl button I get the error that I titled this report.  "INVALID GPU OR INVALID FILES IN FOLDER - FAILED". 

I've restarted multiple times and it persists.
I've had no problem upscaling any of these images when I do them individually.
I converted the images independently from PNG to JPG and the problem persisted.

### To Reproduce

Without sending you my files, I'm not sure.

### Version (or commit hash)

2.9.8

### Did you download Upscayl from the Mac App Store?

- [ ] Yes, I downloaded Upscayl from the Mac App Store!

### Select your Operating System

MacOS

### OS Version

MacOS Sonoma 14.2.1

### GPU Name

M1 16GB

### Screenshots

![Screenshot 2024-01-24 at 4 29 11 PM](https://github.com/upscayl/upscayl/assets/12870435/537ac798-4508-410a-9053-2ab0f806cafc)

### Paste the logs

```sh
🧠 Recalling Output Folder:  /Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🔀 Setting model to realesrgan-x4plus
🐞 BACKEND REPORTED:  📁 Updating Output Folder Path:  /Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran
🐞 BACKEND REPORTED:  💾 Updating Save Output Folder:  true
🐞 BACKEND REPORTED:  📐 Updating Compression:  0
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
⚙️ Getting rememberOutputFolder from localStorage:  true
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🔀 Setting model to realesrgan-x4plus
🔄 Resetting image paths
🐞 BACKEND REPORTED:  📁 Updating Folder Path:  /Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran/Originals/New Folder With Items 2/New Folder With Items
🖼 Selected Folder Path:  /Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran/Originals/New Folder With Items 2/New Folder With Items
🐞 BACKEND REPORTED:  📁 Selected Folder Path:  /Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran/Originals/New Folder With Items 2/New Folder With Items
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
⚙️ Getting rememberOutputFolder from localStorage:  true
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🔀 Setting model to realesrgan-x4plus
🔄 Resetting Upscaled Image Path
🏁 FOLDER_UPSCAYL
🐞 BACKEND REPORTED:  🖼️ Updating No Image Processing:  false
🐞 BACKEND REPORTED:  📐 Updating Compression:  0
🐞 BACKEND REPORTED:  📢 Upscayl Command:  -i,/Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran/Originals/New Folder With Items 2/New Folder With Items,-o,/Volumes/Prometheus/Downloads/New Folder With Items 2/Poldaran/Originals/New Folder With Items 2/New Folder With Items/upscayl_realesrgan-x4plus_x4,-s,4,-m,/Applications/Upscayl.app/Contents/Resources/models,-n,realesrgan-x4plus,-f,png
🐞 BACKEND REPORTED:  🛑 Updating Stopped:  false
🔄 Resetting image paths
🚧 FOLDER_UPSCAYL_PROGRESS:  filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 01.png
good filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 01.png
filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 02.png
good filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 02.png
filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 03.png
good filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 03.png
filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 04.png
good filename: closeup of failed spork prototypes in the engineering bay of an android starship, white walls and floors with soft white lighting - 04.png

🐞 BACKEND REPORTED:  ❌ INVALID GPU OR INVALID FILES IN FOLDER - FAILED
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
⚙️ Getting rememberOutputFolder from localStorage:  true
```

**Comment by aaronliu0130** [comment]
Does the folder only contain images?

**Comment by kriebe** [comment]
Yes, only the 4 images and nothing else.  It happens with other images too, but not always.

**Comment by kriebe** [comment]
I didn't bother showing invisible, but that's because I've done it with multiple folders and moved the 4 pictures into other folders and still had the problem with them.  And to be clear, it wasn't just those 4.  There were other images in the original folder they were in that also didn't work and I ended up having to use something else to upscale them instead of doing them individually.  But since those images, the problem has not resurfaced.  I wouldn't have thought twice about it, but since I even created new versions of the images and saved them as different types (PNG to JPG) and the problem persisted, I felt I should share it with you.

**Comment by NayamAmarshe** [comment]
Try renaming the files to something shorter maybe

**Comment by pongraczi** [comment]
Try these:  
- do not use space in the filename or path, use underscore (I did not try to proof, but your filenames seems have white spaces in the name)
- if you have any non-ascii characters in the path or name (like: éáűúőóüöí etc.), change it to eauoi etc., (I did and upscayl started working -> I got results, saved as expected)

**Comment by kriebe** [comment]
spaces have zero effect whatsoever.  I've upscaled thousands of files and it's never mattered.  It's been a while and the issue still happens with a few files that I created around that time, no matter the naming.  I never use special characters either.  Unless you consider a comma or a dash a special character, but again, I've used those thousands of times with zero issues.  Anyway, I'm not getting the problem anymore, unless I go back to those same files.  And again, no matter what I name them or even convert them to a different format, they still don't work.  I'm not worried about it anymore.
